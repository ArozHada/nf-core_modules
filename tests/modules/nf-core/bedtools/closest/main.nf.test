nextflow_process {

    name "Test Process BEDTOOLS_CLOSEST"
    script "modules/nf-core/bedtools/closest/main.nf"
    process "BEDTOOLS_CLOSEST"
    config "tests/modules/nf-core/bedtools/closest/nextflow.config"
    tag "bedtools_closest"

    test("Human Genome Bed") {

        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = [
                             [ id:'test' ], // meta map
                             file(params.test_data['homo_sapiens']['genome']['genome_bed'], checkIfExists: true),
                             [
                              file(params.test_data['homo_sapiens']['genome']['genome_multi_interval_bed'], checkIfExists: true),
                              file(params.test_data['homo_sapiens']['genome']['genome_21_multi_interval_bed'], checkIfExists: true)
                             ]
                           ]
                input[1] = []
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(path(params.outdir).list()).match("genome_bed") }
            )
        }
    }

    test("Human Genome Bed With Fasta") {

        options "--sorted"

        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = [
                            [ id:'test' ], // meta map
                            file(params.test_data['homo_sapiens']['genome']['genome_bed'], checkIfExists: true),
                            file(params.test_data['homo_sapiens']['genome']['genome_multi_interval_bed'], checkIfExists: true)
                           ]
                input[1] = file(params.test_data['homo_sapiens']['genome']['genome_fasta_fai'], checkIfExists: true)
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(path(params.outdir).list()).match("genome_bed_fasta") }
            )
        }
    }

    test("Human Genome VCF") {

        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = [
                            [ id:'test', single_end:false ], // meta map
                            file(params.test_data['homo_sapiens']['genome']['dbsnp_146_hg38_vcf_gz'], checkIfExists: true),
                            file(params.test_data['homo_sapiens']['genome']['gnomad_r2_1_1_vcf_gz'], checkIfExists: true)
                           ]
                input[1] = []
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(path(params.outdir).list()).match("vcf") }
            )
        }
    }
}
