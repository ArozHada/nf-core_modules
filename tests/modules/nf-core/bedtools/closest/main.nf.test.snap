{
    "genome_bed_fasta": {
        "content": [
            [
                "test.bed:md5,347f192c141a34dfc4bb7c13fffd44bb",
                "versions.yml:md5,21b18c873c7ca8f5166982532fd82266"
            ]
        ],
        "timestamp": "2023-04-11T19:42:32+0000"
    },
    "genome_bed": {
        "content": [
            [
                "test.bed:md5,de47864c0ae53891d09252eb63cd7269",
                "versions.yml:md5,21b18c873c7ca8f5166982532fd82266"
            ]
        ],
        "timestamp": "2023-04-11T19:42:32+0000"
    },
    "vcf": {
        "content": [
            [
                "test.vcf:md5,376bb3583812c2ffd354510f82f55bc7",
                "versions.yml:md5,21b18c873c7ca8f5166982532fd82266"
            ]
        ],
        "timestamp": "2023-04-11T19:42:32+0000"
    }
}